File: samples/CSharp/SourceGenerators/GeneratedDemo/UseResxSourceGenerator.cs
using System;

namespace GeneratedDemo
{
    internal static class UseResxSourceGenerator
    {
        public static void Run()
        {
            Console.WriteLine($"Reading resources from '{typeof(global::GeneratedDemo.Resources)}'");
            Console.WriteLine(global::GeneratedDemo.Resources.Example);

            Console.WriteLine($"Reading resources from '{typeof(global::GeneratedDemo.SubFolder.Resources)}'");
            Console.WriteLine(global::GeneratedDemo.SubFolder.Resources.SubExample);
        }
    }
}

File: samples/CSharp/SourceGenerators/GeneratedDemo.VisualBasic/UseResxSourceGenerator.vb
Friend Module UseResxSourceGenerator
    Public Sub Run()
        Console.WriteLine($"Reading resources from '{GetType(Global.GeneratedDemo.VisualBasic.Resources)}'")
        Console.WriteLine(Global.GeneratedDemo.VisualBasic.Resources.Example)

        Console.WriteLine($"Reading resources from '{GetType(Global.GeneratedDemo.VisualBasic.SubFolder.Resources2)}'")
        Console.WriteLine(Global.GeneratedDemo.VisualBasic.SubFolder.Resources2.SubExample)
    End Sub
End Module
